containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -148,6 +148,38 @@ struct UnorderedMapInsertOpTypes {
   };
 };
 
+namespace Impl {
+
+// Functor for UnorderedMap to set its flag
+template <typename View>
+struct FlagSetFunctor {
+  View m_view;
+  bool m_value;
+
+  FlagSetFunctor(View const &view, bool const value)
+      : m_view(view), m_value(value) {}
+
+  KOKKOS_FUNCTION
+  void operator()(unsigned const i) const {
+    m_view(i) = static_cast<int>(m_value);
+  }
+};
+
+// Functor for UnorderedMap to get its flag
+template <typename View>
+struct FlagGetFunctor {
+  View m_view;
+
+  explicit FlagGetFunctor(View const &view) : m_view(view) {}
+
+  KOKKOS_FUNCTION
+  void operator()(unsigned const i, bool &value_inner) const {
+    value_inner = static_cast<bool>(m_view(i));
+  }
+};
```

Comment:
These are so short that it hsould be fine to inline them.

Reply:
Do you mean using `inline`?

Reply:
Eventually replaced it by a call to `deep_copy` + a scalar subview.